packages/enzyme/src/ShallowWrapper.js

```diff
@@ -310,78 +311,58 @@ class ShallowWrapper {
         const { state } = instance;
         const prevProps = instance.props || this[UNRENDERED].props;
         const prevContext = instance.context || this[OPTIONS].context;
-        const nextProps = { ...prevProps, ...props };
         const nextContext = context || prevContext;
         if (context) {
           this[OPTIONS] = { ...this[OPTIONS], context: nextContext };
         }
         this[RENDERER].batchedUpdates(() => {
+          // When shouldComponentUpdate returns false we shouldn't call componentDidUpdate.
+          // so we spy shouldComponentUpdate to get the result.
           let shouldRender = true;
-          // dirty hack:
-          // make sure that componentWillReceiveProps is called before shouldComponentUpdate
```

Comment:
do we no longer need this hack?

Reply:
Can you elaborate on why?

Reply:
The current version calls `shouldComponentUpdate` manually to get the result.
But `componentWillReceiveProps` should be called before `shouldComponentUpdate` so we should call `componentWillReceiveProps` manually before `shouldComponentUpdate`.

In this PR, we no longer call `shouldComponentUpdate` manually because we can spy the method.
As the result, we can remove the dirty hack because we don't need to call `componentWillReceiveProps` and `shouldComponentUpdate` manually.

Reply:
Perfect, thanks.